pkg/types/upstream.go

```diff
@@ -445,3 +441,16 @@ type SortedPair struct {
 	Key   string
 	Value string
 }
+
+func init() {
+	ConnPollFactories = make(map[Protocol]connPoolFactory)
+}
+
+type connPoolFactory func(lbCtx LoadBalancerContext, clusterName string) ConnectionPool
+
+var ConnPollFactories map[Protocol]connPoolFactory
```

Comment:
do not understand why it is need connPoolFactory.
All protocol register a nil,  and no one use the registered connPoolFactory, looks like just need a bool?